### PR TITLE
ARROW-3383: [Gandiva][Java] Fix java build

### PR DESCRIPTION
**Pull request by pravindra**
- set arrow as parent project, use same style as arrow java
- fix lots of checkctyle violations
- add java build/tests to CI

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -93,21 +93,21 @@ matrix:
...
     - ARROW_TRAVIS_VALGRIND=1
     - ARROW_TRAVIS_CLANG_FORMAT=1
     - ARROW_BUILD_WARNING_LEVEL=CHECKIN
-    - ARROW_TRAVIS_GANDIVA=1
     - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
     before_script:
-    # Run if something changed in CPP.
-    - if [ $ARROW_CI_CPP_AFFECTED != "1" ]; then exit; fi
+    # Run if something changed in CPP or Java.
+    - if [ $ARROW_CI_CPP_AFFECTED != "1" && $ARROW_CI_JAVA_AFFECTED != "1" ]; then exit; fi
```
Comment (praveenbingo) [inline .travis.yml]: Or?

Comment (praveenbingo) [reply .travis.yml]: never mind..read it wrong the first time..

**Comment by wesm** [comment]
Just rebased, so the build should pass now (unless we hit ARROW-3343)

**Review by wesm — APPROVED** [review]
+1. Will merge once the build runs

**Comment by wesm** [comment]
Does this have any conflict with https://github.com/apache/arrow/pull/2717? I should probably merge that first

**Comment by pravindra** [comment]
No conflict with #2717 but it’s likely that will show up checkstyle errors (this PR switches to arrow checkstyle) in the new java code. Please merge that one first. I’ll fix the checkstyle errors.

**Comment by wesm** [comment]
OK, merged #2717

**Comment by wesm** [comment]
I had forgotten to push the Gandiva commits to apache/master. Should be good now

**Comment by pravindra** [comment]
@wesm can you please merge now ?

**Comment by wesm** [comment]
Yep! merging now

**Pull request closed without merging**